src/libsystemd/sd-event/sd-event.c

```diff
@@ -706,6 +706,9 @@ static void event_unmask_signal_data(sd_event *e, struct signal_data *d, int sig
                 return;
         }
 
+        if (event_pid_changed(e))
```

Comment:
hm, i am not convinced we shoud check for that here. I mean, it's not supported to use sd-event across forks(), and we generate an error on most entry points already.

I figure we should just change source_free() and event_free() to start with a line like this:

```c
assert_return(!event_pid_changed(e), NULL);
```

this means we'll somewhat gracefully handle attempts to free event or event source objects allocated in children: we just ignore the request and debug log about it.